numba/_helperlib.c

```diff
@@ -196,6 +196,18 @@ numba_signbit(npy_double a)
     return npy_signbit(a);
 }
 
+NUMBA_EXPORT_FUNC(float)
+numba_cbrtf(float x)
+{
+    return npy_cbrtf(x);
+}
+
+NUMBA_EXPORT_FUNC(double)
+numba_cbrt(double x)
+{
+    return npy_cbrt(x);
+}
```

Comment:
I'm struggling to understand why this is needed?

This:
```python
from numba import njit
import numpy as np

@njit(fastmath={'nnan', 'nsz', 'ninf', 'afn'})
def cbrt(x):
    third = 1.0 / 3.0
    return np.power(x, third)

@njit
def foo(x):
    return cbrt(x)

x = np.zeros(100)
foo(x)
print(cbrt.inspect_llvm(cbrt.signatures[0]))

print(cbrt.inspect_asm(cbrt.signatures[0]))
```
generates a call to the `cbrt` function, LLVM "knows" that `power(x, 1/3)` is `cbrt` under the flags set. Perhaps this can be used to ensure that a `cbrt` call is generated along with adding the inf/nan handling as per NumPy?

Reply:
> I'm struggling to understand why this is needed?
Before adding this, one of the tests were failing on CI. See:
https://github.com/numba/numba/pull/6075#discussion_r482623254

> generates a call to the cbrt function, LLVM "knows" that power(x, 1/3) is cbrt under the flags set.

For me, this produces a call to `@llvm.pow.f64(x, 1/3)`:
```llvm
; Function Attrs: nofree nounwind writeonly
define i32 @"_ZN8__main__8cbrt$242Ex"(double* noalias nocapture %retptr, { i8*, i32, i8* }** noalias nocapture readnone %excinfo, i64 %arg.x) local_unnamed_addr #0 {
entry:
  %.37.le = sitofp i64 %arg.x to double
  %.38.le = tail call nnan ninf nsz afn double @llvm.pow.f64(double %.37.le, double 0x3FD5555555555555)
  store double %.38.le, double* %retptr, align 8
  ret i32 0
}
```

Reply:
Yes, but that then ends up as a call to `cbrt` once compiled?

Reply:
Oh yes, you right. Nevertheless, If I remove the code you quote, some builds on CI starts to fail. Do you want me to give a try and implement `cbrt` in numba?

Reply:
If one enables fast-math, you'll see a call to cbrt.

```python
from numba import njit
import numpy as np

@njit('float32(float32)', fastmath=True, no_cfunc_wrapper=True, no_cpython_wrapper=True)
def foo(x):
    return np.cbrt(x)

print(foo.inspect_asm(foo.signatures[0]))
```

```asm
	.section	__TEXT,__text,regular,pure_instructions
	.macosx_version_min 10, 15
	.section	__TEXT,__literal4,4byte_literals
	.p2align	2
LCPI0_0:
	.long	2147483648
	.section	__TEXT,__text,regular,pure_instructions
	.globl	__ZN8__main__7foo$241Ef
	.p2align	4, 0x90
__ZN8__main__7foo$241Ef:
	pushq	%r14
	pushq	%rbx
	subq	$56, %rsp
	vmovaps	%xmm0, (%rsp)
	movq	%rdi, %rbx
	vcvtss2sd	%xmm0, %xmm0, %xmm0
	movabsq	$_cbrt, %r14
	callq	*%r14
	vcvtsd2ss	%xmm0, %xmm0, %xmm0
	vmovaps	%xmm0, 32(%rsp)
	movabsq	$LCPI0_0, %rax
	vbroadcastss	(%rax), %xmm0
	vmovaps	%xmm0, 16(%rsp)
	vxorps	(%rsp), %xmm0, %xmm0
	vcvtss2sd	%xmm0, %xmm0, %xmm0
	callq	*%r14
	vcvtsd2ss	%xmm0, %xmm0, %xmm0
	vxorps	16(%rsp), %xmm0, %xmm0
	vxorps	%xmm1, %xmm1, %xmm1
	vmovaps	(%rsp), %xmm2
	vcmpltss	%xmm1, %xmm2, %xmm1
	vmovaps	32(%rsp), %xmm2
	vblendvps	%xmm1, %xmm0, %xmm2, %xmm0
	vmovss	%xmm0, (%rbx)
	xorl	%eax, %eax
	addq	$56, %rsp
	popq	%rbx
	popq	%r14
	retq

	.comm	__ZN08NumbaEnv8__main__7foo$241Ef,8,3
	.comm	__ZN08NumbaEnv5numba2np8npyfuncs17np_real_cbrt_impl12$3clocals$3e8cbrt$242Ef,8,3
.subsections_via_symbols
```